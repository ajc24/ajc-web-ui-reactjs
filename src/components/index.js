/**
 * Developed by Anthony Cox in 2024
 */

/* Export the header components */
export { default as HeaderTypeA } from './header/HeaderTypeA';

/* Export the main components */
export { default as MainContent } from './main/MainContent';

/* Export the menu bar components */
export { default as MenuBarTypeA } from './menu-bar/MenuBarTypeA';
