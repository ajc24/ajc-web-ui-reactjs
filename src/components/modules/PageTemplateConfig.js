/**
 * Developed by Anthony Cox in 2025
 */
import { storybookCssClassList } from '../data/storybook';

/* Commonly used CSS classes */
const bodyDefaultCss = 'body-default';

/**
 * Functionality designed to work with the page template components (headers, menu bars, body content, footers)
 */
export default class PageTemplateConfig {
  /**
   * Setup and configure the current web page to ensure the default CSS styling is set to it.
   * This also handles the styling of pages which are deployed on Storybook
   */
  static setupDocumentBodyCss() {
    /* If we are rendering on Storybook, remove the Storybook classes which will affect the width of the component */
    for (let index = 0; index < storybookCssClassList.length; index += 1) {
      if (document.body.classList.contains(storybookCssClassList[index]) === true) {
        document.body.classList.remove(storybookCssClassList[index]);
      }
    }
    /* Ensure that the <body> element stretches to 100% of the width of the screen, that the X axis scrollbar is hidden and Y axis scrollbar is always displayed */
    if (document.body.classList.contains(bodyDefaultCss) === false) {
      document.body.classList.add(bodyDefaultCss);
      document.activeElement.blur();
    }
  }
}
