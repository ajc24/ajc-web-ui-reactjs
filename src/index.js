/**
 * Developed by Anthony Cox in 2024
 */

/* Import the header components */
import Header from './components/header/Header';

/* Import the image components */
import DecorativeImage from './components/images/DecorativeImage';
import FunctionalImage from './components/images/FunctionalImage';
import InformativeImage from './components/images/InformativeImage';

/* Export all components */
export {
  DecorativeImage,
  FunctionalImage,
  InformativeImage,
  Header,
};
