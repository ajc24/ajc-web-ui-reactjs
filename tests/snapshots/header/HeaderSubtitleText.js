/**
 * Developed by Anthony Cox in 2025
 */
import 'jsdom-global/register';
import React from 'react';
import { TestDev } from 'ajc-testing-code';
import { HeaderSubtitleText } from '../../../src';
import PageTemplateConfig from '../../../src/components/modules/PageTemplateConfig';

describe('Header Subtitle Text', () => {
  /* Set the data for the tests */
  const testComponentId = 'test-component-id';

  /* Set the spies for use in the tests */
  let setupDocumentBodyCssSpy;

  beforeAll(() => {
    setupDocumentBodyCssSpy = jest
      .spyOn(PageTemplateConfig, 'setupDocumentBodyCss')
      .mockImplementation(() => {});
  });

  afterAll(() => {
    setupDocumentBodyCssSpy.mockRestore();
  });

  describe('Default standalone component, left alignment, no title text rendered, black text colour, marked as visible', () => {
    let getBoundingClientRectSpy;
    let snapshot;
    
    beforeAll(() => {
      /* Create the spies for this component test */
      getBoundingClientRectSpy = jest
        .spyOn(Element.prototype, 'getBoundingClientRect')
        .mockImplementationOnce(() => {
          /* setRightmostSubtitleTextPosition() - setting paragraphRightPos */
          return {
            right: 500,
          };
        })
        .mockImplementationOnce(() => {
          /* handleVisibility() - setting containerRightPos */
          return {
            right: 500,
          };
        });
      
      /* Generate the snapshot for the component */
      snapshot = TestDev.createSnapshot(
        <React.Fragment>
          <HeaderSubtitleText id={testComponentId} alignment="left" textColour="black">
            Header subtitle text component text content.
          </HeaderSubtitleText>
        </React.Fragment>
      );
    });

    afterAll(() => {
      getBoundingClientRectSpy.mockRestore();
    });

    it('verifies the snapshot for the component', () => {
      expect(snapshot).toMatchSnapshot();
    });
  });

  describe('Default standalone component, centre alignment, no title text rendered, white text colour, marked as hidden', () => {
    let getBoundingClientRectSpy;
    let snapshot;
    
    beforeAll(() => {
      /* Create the spies for this component test */
      getBoundingClientRectSpy = jest
        .spyOn(Element.prototype, 'getBoundingClientRect')
        .mockImplementationOnce(() => {
          /* setRightmostSubtitleTextPosition() - setting paragraphRightPos */
          return {
            right: 501,
          };
        })
        .mockImplementationOnce(() => {
          /* handleVisibility() - setting containerRightPos */
          return {
            right: 500,
          };
        });
      
      /* Generate the snapshot for the component */
      snapshot = TestDev.createSnapshot(
        <React.Fragment>
          <HeaderSubtitleText id={testComponentId} alignment="centre" textColour="white">
            Header subtitle text component text content.
          </HeaderSubtitleText>
        </React.Fragment>
      );
    });

    afterAll(() => {
      getBoundingClientRectSpy.mockRestore();
    });

    it('verifies the snapshot for the component', () => {
      expect(snapshot).toMatchSnapshot();
    });
  });

  describe('Component rendered inside Header, default alignment, with title text rendered, default text colour, marked as visible', () => {
    let getBoundingClientRectSpy;
    let snapshot;
    
    beforeAll(() => {
      /* Create the spies for this component test */
      getBoundingClientRectSpy = jest
        .spyOn(Element.prototype, 'getBoundingClientRect')
        .mockImplementation(() => { return { right: 0 } });
      
      /* Generate the snapshot for the component */
      snapshot = TestDev.createSnapshot(
        <React.Fragment>
          <HeaderSubtitleText id={testComponentId} headerTitleTextId="test-header-title-text-id" parentHeaderId="test-parent-header-id">
            Header subtitle text component text content.
          </HeaderSubtitleText>
        </React.Fragment>
      );
    });

    afterAll(() => {
      getBoundingClientRectSpy.mockRestore();
    });

    it('verifies the snapshot for the component', () => {
      expect(snapshot).toMatchSnapshot();
    });
  });
});
