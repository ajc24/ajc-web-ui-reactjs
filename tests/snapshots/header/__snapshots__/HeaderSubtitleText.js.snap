// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Subtitle Text Component rendered inside Header, default alignment, with title text rendered, default text colour, marked as visible verifies the snapshot for the component 1`] = `"<div id="test-component-id--header-subtitle-text" class="background-transparent subtitle-text-size-inside-header subtitle-text-alignment-left subtitle-text-with-header-text"><p class="subtitle-text-default font-black" aria-hidden="false">Header subtitle text component text content.</p></div>"`;

exports[`Header Subtitle Text Default standalone component, centre alignment, no title text rendered, white text colour, marked as hidden verifies the snapshot for the component 1`] = `"<div id="test-component-id--header-subtitle-text" class="background-transparent page-template-container subtitle-text-alignment-centre"><p class="subtitle-text-default font-white hidden" aria-hidden="true">Header subtitle text component text content.</p></div>"`;

exports[`Header Subtitle Text Default standalone component, left alignment, no title text rendered, black text colour, marked as visible verifies the snapshot for the component 1`] = `"<div id="test-component-id--header-subtitle-text" class="background-transparent page-template-container subtitle-text-alignment-left"><p class="subtitle-text-default font-black" aria-hidden="false">Header subtitle text component text content.</p></div>"`;
